azerothcore/azerothcore-wotlk
# feat(Core/Spells): improve SpellCheckRange(), spell cast when dismounting, falling or levitating spells

imported Maczuga/SunwellCore@5d9826c

## CHANGES PROPOSED:
-  enable item use and spell cast when dismounting.
- improve SpellCheckRange()
- spell cast when dismounting
- falling or levitating spells
- Make certain spells (e.g. Mind Control) properly flag the caster for PvP

## ISSUES ADDRESSED:
- related #1141

## HOW TO TEST THE CHANGES:
I am not sure about the steps to tests these changes, but I can figure out that the followings step can help:
- use the levitate spell of Priest and the mage one for levitating
- try to fall from a high place
- try to fall dismounting from a flying mount
- try to fall dismounting when you are jumping from a high place and dismounting and use spells
- cast spells while are you in mount
- try to use spells (Melle and Ranged, so use Warrior, Hunter and Mage maybe) and use from very far (also if the client show that you can't) or from near NPCs
- use Mind Control in duel against other player and on NPC, use it once in battleground, check if everything seems ok

## Target branch(es):








 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
